---
title: How to visualise phylogenetic trees and alignments
permalink: /docs/howto_Visualize_Data/
redirect_from: /index.html
---

### Visualising Multi-Sequence-Alignments using Aliview

Discovering nucleotide/amino-acid variations in a MSA can be tricky. One way to help the user to overcome
this issue is using dedicated softwares that allow basic operations on the MSA object (i.e. alignment browsing,
coloring, indexing and nucleotide/amino-acid translation).

#### 1. Open a MSA file

<center><img src="../../img/vis1.png" alt="" title="" width="100%"></center>

#### 2. Perform a view

<center><img src="../../img/vis2.png" alt="" title="" width="100%"></center>

#### 3. Extract view

<center><img src="../../img/vis3.png" alt="" title="" width="100%"></center>

#### 4. Summary statistics on the view

<center><img src="../../img/vis4.png" alt="" title="" width="100%"></center>

#### 5. Translate a selection view using different codon frames

<center><img src="../../img/vis5.png" alt="" title="" width="100%"></center>

---
#### Other visualization softwares
You can find a list of MSA visualizers here:  https://en.wikipedia.org/wiki/List_of_alignment_visualization_software

---

### Visualizing phylogenetic trees with FigTree

<center><img src="../../img/FigTree_1.jpg" alt="" title="" width="100%"></center>
